### PR TITLE
[extension-selenium] Add step to check element existence for duration

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericWaitSteps.java [hunk 1]
```diff
@@ -59,6 +66,22 @@ public void waitForElementDisappearance(Locator locator)
         waitForCondition(expectedSearchActionsConditions.invisibilityOfElement(locator));
     }
 
+    /**
+     * Checks that element located by <b>locator</b> exists during the <b>duration</b>
+     * @param locator locator to find an element
+     * @param duration total waiting time according to <a href="https://en.wikipedia.org/wiki/ISO_8601">ISO 8601</a>
+     * standard
+     */
+    @Then("element located '$locator' exists for '$duration' duration")
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericWaitSteps.java:75]:
```
Then element located `$locator` exists for `$duration` duration
```

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericWaitSteps.java [hunk 2]
```diff
@@ -59,6 +66,22 @@ public void waitForElementDisappearance(Locator locator)
...
+    public void doesElementByLocatorExistsForDuration(Locator locator, Duration duration)
+    {
+        WaitResult<Boolean> result = waitActions.wait(uiContext.getSearchContext(), duration,
+                expectedSearchActionsConditions
+                        .not(expectedSearchActionsConditions.presenceOfAllElementsLocatedBy(locator)), false);
+        softAssert.assertFalse(String.format("Element located by locator '%s' has existed during '%s' duration",
+                locator, duration.toSeconds()), result.isWaitPassed());
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericWaitSteps.java:82]: is there any way to pretty-print duration?

File: vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/GenericWaitStepsTests.java [hunk 3]
```diff
@@ -75,4 +81,26 @@ void testWaitTillElementDisappears()
...
+        waitResult.setWaitPassed(true);
+
+        when(uiContext.getSearchContext()).thenReturn(webElement);
+        when(expectedSearchActionsConditions.presenceOfAllElementsLocatedBy(locator)).thenReturn(presenceOfAllElements);
+        when(expectedSearchActionsConditions.not(presenceOfAllElements)).thenReturn(not);
+        when(waitActions.wait(webElement, Duration.ZERO, not, false)).thenReturn(waitResult);
+
+        waitSteps.doesElementByLocatorExistsForDuration(locator, Duration.ZERO);
+
+        verify(softAssert).assertFalse("Element located by locator ' Search: 'value'; Visibility: VISIBLE;' has existed"
```
Comment (valfirst) [inline vividus-extension-selenium/src/test/java/org/vividus/bdd/steps/ui/GenericWaitStepsTests.java:103]: I believe we should revise `org.vividus.ui.action.search.Locator#toString` as first

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/WaitSteps.java [hunk 4]
```diff
@@ -338,9 +338,11 @@ public void elementByIdDisappears(String id)
 
     /**
      * Checks that element exists during the timeout
+     * @deprecated Use <code>Then element located '$locator' exists for '$duration' duration</code>
      * @param seconds timeout in seconds
      * @param xpath XPath value of the element
      */
+    @Deprecated(forRemoval = true, since = "0.3.0")
```
Comment (valfirst) [inline vividus-plugin-web-app/src/main/java/org/vividus/bdd/steps/ui/web/WaitSteps.java:345]: `since = "0.2.9"`

**Review by ikalinin1 — APPROVED** [review]

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericWaitSteps.java [hunk 5]
```diff
@@ -59,6 +68,41 @@ public void waitForElementDisappearance(Locator locator)
...
+                locator.toHumanReadableString(), prettyDuration);
+        softAssert.assertFalse(assertionMessage, result.isWaitPassed());
+    }
+
+    private String formatDuration(Duration duration)
+    {
+        String durationFormat = "";
+        if (duration.toSeconds() != 0)
+        {
+            durationFormat += "s' seconds' ";
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericWaitSteps.java:96]: what about adding minutes as well?

File: vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericWaitSteps.java [hunk 6]
```diff
@@ -59,6 +68,41 @@ public void waitForElementDisappearance(Locator locator)
...
+    @Then("element located `$locator` exists for `$duration` duration")
+    public void doesElementByLocatorExistsForDuration(Locator locator, Duration duration)
+    {
+        String prettyDuration = formatDuration(duration);
+
+        WaitResult<Boolean> result = waitActions.wait(uiContext.getSearchContext(), duration,
+                expectedSearchActionsConditions
+                        .not(expectedSearchActionsConditions.presenceOfAllElementsLocatedBy(locator)), false);
+
+        String assertionMessage = String.format("Element located by locator %s has existed during %s",
```
Comment (valfirst) [inline vividus-extension-selenium/src/main/java/org/vividus/bdd/steps/ui/GenericWaitSteps.java:86]: `during` vs `for`?

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**